pybamm/expression_tree/broadcasts.py

```diff
@@ -107,14 +118,10 @@ def evaluate_for_shape(self):
         """
         child_eval = self.children[0].evaluate_for_shape()
         vec = pybamm.evaluate_for_shape_using_domain(self.domain)
-
-        if self.broadcast_type == "primary":
-            return np.outer(child_eval, vec).reshape(-1, 1)
-        elif self.broadcast_type == "full":
-            return child_eval * vec
+        return np.outer(child_eval, vec).reshape(-1, 1)
 
 
-class PrimaryBroadcast(Broadcast):
+class SecondaryBroadcast(Broadcast):
     """A node in the expression tree representing a primary broadcasting operator.
     Broadcasts in a `primary` dimension only. That is, makes explicit copies
```

Comment:
"`secondary` dimension only", and as above the last part of this isn't clear to me